src/postprocess/DOS.jl

```diff
@@ -0,0 +1,41 @@
+# Compute densities of states
+
+# IDOS (integrated)
+# N(ε) = sum_n f_n = sum_n f((εn-ε)/T)
+# DOS
+# D(ε) = N'(ε)
+# LDOS (local)
+# LD = sum_n f_n ψn
+
+using ForwardDiff
+
+"""
+Total density of states at energy ε
+"""
+function DOS(ε, basis, orben; smearing=basis.model.smearing, T=basis.model.temperature)
+    D = zero(ε)
+    T != 0 || error("DOS only supports finite temperature")
+    for ik = 1:length(orben)
+        for iband = 1:length(orben[ik])
+            D += basis.kweights[ik]*ForwardDiff.derivative(ε -> smearing((orben[ik][iband]-ε)/T), ε)
+        end
+    end
+    D
+end
+
+"""
+Local density of states, in real space
+"""
+function LDOS(ε, basis, orben, Psi; smearing=basis.model.smearing, T=basis.model.temperature)
+    D = zeros(typeof(ε), basis.fft_size)
```

Comment:
I'm not sure `typeof(ε)` is the right expression to deduce the type here. Don't you think `real(eltype(Psi[1]))` or `eltype(orben[1])` is better?

Reply:
I'm intensely relaxed about that. If the user wants to mix precision they're responsible for ensuring everything is in the proper type. Eps is as good as anything here, but I can change it if you don't like it

Reply:
Usually I used `real(eltype(Psi[1]))`, so for consistency I'd prefer that.

Reply:
But then it's not consistent with DOS... but sure, why not

Reply:
hmmm ... true thing.